### ISSUE TITLE
Offer multilingual support

### ISSUE THREAD

**Issue by lwcorp** [bramley/phplist-plugin-viewbrowser#24]
phpList itself has multilingual support, so possibly integrate with it.
Alternatively, just add a standalone internal multilingual support.
Either way, it would be awesome if all the titles, errors, etc. would be in the admin's chosen language.

**Comment by bramley** [comment]
See https://resources.phplist.com/plugin/viewinbrowser#translations

**Comment by lwcorp** [comment]
Thanks! Are those languages determined by the admin's chosen language for phpList?

**Comment by bramley** [comment]
Not sure quite what you are asking but the answer is probably yes.

**Comment by lwcorp** [comment]
I meant does it base itself on the user's browser (i.e. automated) or the admin's choice (i.e. manual).

**Comment by bramley** [comment]
The admin's selected language is used for the translations that apply to the admin interface, such as the captions on the Settings page.

The translations for public pages, such as archive, don't work as I remembered. phplist will use the browser's preferred language if that language has been installed. I thought that the default front-end language from config.php such as this, would be used but it isn't

    $language_module = 'spanish.inc';

I suppose either is ok if most of the subscribers and visitors are from the same country as the phplist installation.

To use the setting from config.php would require some rework but there are a few real bits of text with a number of validation messages that would happen only if someone was trying to view arbitrary campaigns or arbitrary user uids. So I don't think that is worthwhile.

**Comment by bramley** [comment]
Just looked at this again to refresh my memory. The captions for the plugin's fields on the Settings page are displayed in the admin's selected language, Spanish in this case

<img width="427" height="231" alt="Image" src="https://github.com/user-attachments/assets/d9723da3-2877-4235-ad2e-2dedcceaf4da" />
If you want those to be displayed in your local language then use one of the files in the lan subdirectory as an example.

But the public pages, view and archive, don't handle translations correctly. The translations have been specified in the same way as those for an admin but they need to use a different approach that phplist has for all public pages - the /lists/texts/*.inc files 
There seems to be just the phrase "Campaign archive" that could be translated, along with some validation error messages but those should occur only if a user or campaign has been deleted, which is probably rare.

**Comment by lwcorp** [comment]
> The captions for the plugin's fields on the Settings page are displayed in the admin's selected language

Do you mean those settings aren't manual, and get changed on their own if the admin switches language?

**Comment by bramley** [comment]
The plugin will use translations, if provided by somebody, for the language selected by the admin. If those haven't been provided then it falls back to English. Only Spanish, Dutch and German have been provided so far.

This is for the admin interface only. As I said, the public pages in the plugin don't handle translations correctly so I think will always display in English.

**Comment by lwcorp** [comment]
I think I've opened it back then thinking of end users. It looks weird and out of place in English in non English websites...

**Comment by bramley** [comment]
A new version of the plugin, 2.13.0, improves the translations of text on the public pages. You can create translations in your local language, see https://resources.phplist.com/plugin/viewinbrowser#translations for guidance. You don't need to translate all the texts, I now think that the error messages are quite ok always being displayed in English.

Text on the public pages is displayed in the language specified by the phplist subscribe page. For a subscriber archive page that is the subscribe page from the row in the user table, and for a list archive it is the default subscribe page.

There is an outstanding problem with the month name, if displayed by the date format, on an Archive page which is displayed either in English or a language specified by the browser. I am still looking at that.

**Comment by bramley** [comment]
There is a new version of the plugin, 2.14.0 which uses the php IntlDateFormatter class to format the month name, if specified, for the campaign sent date on the Archive page. The plugin uses the Date format field from the Settings page to format the date.

 I hope that works for any language supported by php. If the php intl extension is not enabled then the plugin uses the core phplist formatDate() function.

**Comment by lwcorp** [comment]
Thanks! What does it mean "if specified"?

By date in settings do you mean the general "date in unix format"?

**Comment by bramley** [comment]
Yes, that date field. what I mean is the month can be displayed as a number (using m or n)  or a short or long name (using M or F respectively). In the latter case it should be displayed in the appropriate language.

**Comment by bramley** [comment]
Front end translations were corrected in release 2.13.0 and improved in release 2.14.0 for the campaign sent date.
